### DISCUSSION TITLE
Where do Azure Devops Yaml files go?

### DISCUSSION THREAD

**Discussion by JeffreyO** [termux/termux-packages#10712] [Q&A]
I am currently writing a few Yaml files to build specific packages into published artifacts by using Azure Pipelines. I'm attempting to aim for configurability without needing to edit the pipeline file, by using Azure Devops variables to pass in a list of packages that need to be built.
Do you accept Yaml files for build automation?

**Comment by MrAdityaAlok** [comment]
> Do you accept Yaml files for build automation?

I do not know much about Azure devops, but what does your yml files do? Is it something similar to github workflow yml file? If yes then answer is no, see [here](https://github.com/termux/termux-packages/wiki/Creating-new-package) to learn about adding new packages to repo. Otherwise please explain further.

**Reply by JeffreyO** [reply]
It's similar, but it doesn't require an API key for pulling everything down and building it. It would also make it easier for anyone to run a custom automated build.
As a bonus, you can also easily run build agents on local machines to build everything locally and automatically on as many computers as you want for free, it just would just require the YAML file to written in a way where parallel package builds are possible. You can run a non-parallel build on Microsoft servers though for free up to 3 hours for private projects, per month. So, it works great for testing out stuff before opening a pull request for the public project.

Here is my example for a single package build for reference. It's not complete, but it works.
[azure-pipeline-termux-build-docker-build-single-package.yml.txt](https://github.com/termux/termux-packages/files/8758780/azure-pipeline-termux-build-docker-build-single-package.yml.txt)

**Reply by MrAdityaAlok** [reply]
If you want to run tests before opening a PR:
- you can either you use Termux docker image or
- enable github worflows in your fork of repo (then on each push, CI will run)
- or even build on Android device

Other than this, we can do nothing. We have no plan of using azure devops.
